packages/application-config/src/process-config.ts

```diff
@@ -28,21 +40,182 @@ type ProcessConfigOptions = Partial<LoadingConfigOptions> & {
+    authorizeUrl: [
+      // In case the MC API url points to localhost, we need to point
+      // to a local running dev login page to handle the workflow properly.
+      mcApiUrl.hostname === 'localhost'
+        ? mcApiUrl.origin.replace(mcApiUrl.port, String(developmentPort))
+        : mcApiUrl.origin.replace('mc-api', 'mc'),
+      '/login/authorize',
+    ].join(''),
+    initialProjectKey:
+      // For the `account` application, we should unset the projectKey.
+      entryPointUriPath === 'account'
+        ? undefined
+        : appConfig.env.development.initialProjectKey,
+    ...(appConfig.env.development?.teamId && {
+      teamId: appConfig.env.development.teamId,
+      // When using a teamId we need to send the actual application of view identifier.
+      ...(isCustomViewData(configurationData)
+        ? { customViewId: configurationData.id }
+        : { applicationId: configurationData.id }),
+    }),
+    oAuthScopes: appConfig.oAuthScopes,
+    additionalOAuthScopes: appConfig?.additionalOAuthScopes,
+  });
+
+  if (isCustomViewData(configurationData)) {
+    const hostUriPath = (appConfig as JSONSchemaForCustomViewConfigurationFiles)
+      .env.development.hostUriPath;
+    const defaultHostUriPath = oidcConfig.initialProjectKey
+      ? `/${oidcConfig.initialProjectKey}/${entryPointUriPath}`
+      : `/${entryPointUriPath}`;
+    const hostUrl = new URL(
+      hostUriPath || defaultHostUriPath,
+      'http://localhost:3001'
+    );
+    return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+      oidc: oidcConfig,
+      customViewConfig: configurationData,
+      customViewHostUrl: hostUrl.href,
+    });
+  }
+
+  return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+    oidc: oidcConfig,
+    menuLinks: {
+      icon: configurationData.icon,
+      ...configurationData.mainMenuLink,
+      submenuLinks: configurationData.submenuLinks,
+    },
+    // @ts-expect-error: the `accountLinks` is not explicitly typed as it's only used by the account app.
+    accountLinks: configurationData.accountLinks,
+  });
+};
+
+const getRuntimeEnvironmentConfig = ({
+  isProd,
+  configurationData,
+  additionalAppEnv,
+  mcApiUrl,
+  cdnUrl,
+  appUrl,
+  appEnvKey,
+  revision,
+  appConfig,
+}: {
+  isProd: boolean;
+  configurationData: CustomApplicationData | CustomViewData;
+  additionalAppEnv: Record<string, unknown>;
+  mcApiUrl: URL;
+  cdnUrl: URL;
+  appUrl: URL;
+  appEnvKey: string;
+  revision: string;
+  appConfig:
+    | JSONSchemaForCustomApplicationConfigurationFiles
+    | JSONSchemaForCustomViewConfigurationFiles;
+}): ApplicationRuntimeEnvironment => {
+  const entryPointUriPath = isCustomViewData(configurationData)
+    ? // When the application acts as the host for Custom Views, there is no real
+      // entry point to be used, therefore we use a special identifier.
+      CUSTOM_VIEW_HOST_ENTRY_POINT_URI_PATH
+    : configurationData.entryPointUriPath;
```

Comment:
There are many places in App Kit that rely in the `entryPointUriPath` field. I tried making it optional but there are too many tricky places to refactor.

Actually if we think about it, the Custom View is rendered as if it was a Custom Application. In particular, in development it does render as a Custom Application and therefore making the `entryPointUriPath` required.

Therefore, we are now using a "special" value for the `entryPointUriPath`: `@@custom-view-host@@`.
We can then leverage this value in the MC API to run some logic specific to Custom Views, together with the `X-Custom-View-Id` header.

Reply:
That is a very good idea 🙌🏽